components/function-controller/README.md

```diff
@@ -0,0 +1,199 @@
+   * [Creating a sample Hello World Function](#creating-a-sample-hello-world-function)
+
+## Prerequisites
+Before you set up the project, install:
+- Knative Build (v0.6.0)
+- Knative Serving (v0.6.1)
+- Istio (v1.0.7)
+
+## Installation
+
+Install the Function Controller to use it for development purposes. Follow these steps to complete the installation:
+
+### Preparation
+
+Export the following environment variables:
+
+| Variable        | Description | Sample value|
+| --------------- | ----------- |--------------|
+| **IMG**   | The full image name the Function Controller will be tagged with.  | `gcr.io/my-project/function-controller` for GCR, `my-user/function-controller` for Docker Hub |
+| **FN_REGISTRY**   | The URL of the container registry Function images will be pushed to. Used for authentication. | `https://gcr.io/` for GCR, `https://index.docker.io/v1/` for Docker Hub |
+| **FN_REPOSITORY** | The name of the container repository Function images will be pushed to.  |`gcr.io/my-project` for GCR, `my-user` for Docker Hub|
+| `FN_NAMESPACE`  | Namespace where sample functions are to be deployed. |
+
+See the example:
+
+```bash
+export IMG=my-docker-user/function-controller
+export FN_REGISTRY=https://index.docker.io/v1/
+export FN_REPOSITORY=my-docker-user
+```
+
+Create the `serverless-system` Namespace you will deploy the Controller to.
+
+```bash
+kubectl create namespace serverless-system
+```
+
+#### Controller configuration
+
+Create the following configuration for the controller. It contains a list of supported runtimes as well as the container
+repository referenced by the `FN_REPOSITORY` environment variable, which we will create a Secret for in the next steps.
+
+```bash
+cat <<EOF | kubectl -n serverless-system apply -f -
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: fn-config
+data:
+  serviceAccountName: function-controller-build
+  dockerRegistry: ${FN_REPOSITORY}
+  runtimes: |
+    - ID: nodejs8
+      dockerFileName: dockerfile-nodejs-8
+    - ID: nodejs6
+      dockerFileName: dockerfile-nodejs-6
+EOF
+```
+
+#### Functions namespace
+
+Create the Namespace defined previously by the `FN_NAMESPACE` environment variable. The demo Function will be deployed
+to it.
+
+```bash
+kubectl create namespace ${FN_NAMESPACE}
+```
+
+#### Funtion runtimes configurations
+
+Functions require Dockerfiles to be created for each supported runtime. The following manifest creates Dockerfiles for
+Node.js runtimes.
+
+```bash
+kubectl apply -n ${FN_NAMESPACE} -f config/dockerfiles.yaml
+```
+
+
+Before creating Functions, it is necessary to create the `registry-credentials` Secret, which contains credentials to
+the Docker registry defined by the **FN_REGISTRY** environment variable. Knative Build uses this Secret to push the images it builds for the deployed Functions.
+it builds for the deployed Functions. The corresponding `function-controller-build` ServiceAccount was referenced
```

Comment:
```suggestion
For the corresponding `function-controller-build` ServiceAccount see the controller configuration in step 3.
```

Reply:
That changes the intended meaning. The ServiceAccount is not created in step 3, just referenced.